test/e2e/framework/metrics/metrics_grabber.go

```diff
@@ -245,24 +245,26 @@ func (g *Grabber) GrabFromControllerManager() (ControllerManagerMetrics, error)
 		return ControllerManagerMetrics{}, fmt.Errorf("kube-controller-manager: %w", MetricsGrabbingDisabledError)
 	}
 
-	var err error
-	var output string
 	podName := g.kubeControllerManager
-	g.waitForControllerManagerReadyOnce.Do(func() {
```

Comment:
we should keep the ReadyOnce but only with
```
		if readyErr := e2epod.WaitForPodsReady(g.client, metav1.NamespaceSystem, podName, 0); readyErr != nil {
```
inside.
And move the 
```
			output, lastMetricsFetchErr = g.getSecureMetricsFromPod(g.kubeControllerManager, metav1.NamespaceSystem, kubeControllerManagerPort)
```
out of the sync.Once

Reply:
thanks, I moved the `WaitForPodsReady` request to be inside `sync.Once` however I kept the PollImmediate to make the call to get stats from the desired pod.

I was hesitant to add

```
			output, lastMetricsFetchErr = g.getSecureMetricsFromPod(g.kubeControllerManager, metav1.NamespaceSystem, kubeControllerManagerPort)
```

without a retry function because there's the chance that the request fails.